tests/assets/test_digital_assets.py

```diff
@@ -0,0 +1,192 @@
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 2
+    tx['transaction']['asset'].update({'divisible': False})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 0
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+
+
+@pytest.mark.usefixtures('inputs')
+def test_validate_bad_asset_transfer(b, user_vk, user_sk):
+    from bigchaindb.util import get_hash_data
+    from bigchaindb.exceptions import AssetIdMismatch
+
+    tx_input = b.get_owned_ids(user_vk).pop()
+    tx = b.create_transaction(user_vk, user_vk, tx_input, 'TRANFER')
+    tx['transaction']['asset']['id'] = 'aaa'
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, user_sk)
+    with pytest.raises(AssetIdMismatch):
+        b.validate_transaction(tx_signed)
+
+
+def test_validate_asset_arguments(b):
+    from bigchaindb.exceptions import AmountError
+
+    with pytest.raises(TypeError):
+        b.create_transaction(b.me, b.me, None, 'CREATE', divisible=1)
+    with pytest.raises(TypeError):
+        b.create_transaction(b.me, b.me, None, 'CREATE', refillable=1)
+    with pytest.raises(TypeError):
+        b.create_transaction(b.me, b.me, None, 'CREATE', updatable=1)
+    with pytest.raises(TypeError):
+        b.create_transaction(b.me, b.me, None, 'CREATE', data='a')
+    with pytest.raises(TypeError):
+        b.create_transaction(b.me, b.me, None, 'CREATE', amount='a')
+    with pytest.raises(AmountError):
+        b.create_transaction(b.me, b.me, None, 'CREATE', divisible=False, amount=2)
+    with pytest.raises(AmountError):
+        b.create_transaction(b.me, b.me, None, 'CREATE', amount=0)
+
+
+@pytest.mark.usefixtures('inputs')
+def test_get_asset_id_create_transaction(b, user_vk):
+    from bigchaindb.assets import get_asset_id
+
+    tx_input = b.get_owned_ids(user_vk).pop()
+    tx_create = b.get_transaction(tx_input['txid'])
+    asset_id = get_asset_id(tx_input['txid'], bigchain=b)
+
+    assert asset_id == tx_create['transaction']['asset']['id']
+
+
+@pytest.mark.usefixtures('inputs')
+def test_get_asset_id_transfer_transaction(b, user_vk, user_sk):
+    from bigchaindb.assets import get_asset_id
+
+    tx_input = b.get_owned_ids(user_vk).pop()
+    # create a transfer transaction
+    tx_transfer = b.create_transaction(user_vk, user_vk, tx_input, 'TRANSFER')
+    tx_transfer_signed = b.sign_transaction(tx_transfer, user_sk)
+    # create a block
```

Comment:
This can be proposed in a new issue maybe part of a refactor of the test suite.
Creating a fixture and changing all unit tests to use it are out of the scope of this issue